### ISSUE TITLE
[homeconnect] Handler was already disposed

### ISSUE THREAD

**Issue by R-Rosenow** [openhab/openhab-addons#17037]
## Expected Behavior
Since i install the latest Openhab Update, there are warnings in my log. 

I can use the Homeconnect devices like there was no problem.

## Current Behavior
`[WARN ] [.core.thing.binding.BaseThingHandler] - Handler HomeConnectOvenHandler of thing homeconnect:oven:api_bridge_at_home:Backofen tried accessing its bridge although the handler was already disposed.`

## Your Environment
openHAB 4.2.0 Release Build
Proxmox Openhabian

**Comment by lsiepel** [comment]
Are you able to provide a debug log and/or steps to reproduce? 
I can't reproduce this yet.

**Comment by Epeios0** [comment]
I got the same warnings in my log after Update to 4.2.0. I unsinstall and reinstall the Binding, restart Openhab 2 times and the warnings are gonne.

**Comment by lsiepel** [comment]
@R-Rosenow can you reproduce this issue?

**Comment by Blizzard26** [comment]
I'm also seeing this issue since upgrading to 4.2.1 (from 4.1). The warning is printed roughly every 4 minutes.

What I also noticed are the following message **during bootup** of OH:
```
2024-08-30 22:23:31.054 [INFO ] [org.openhab.core.Activator          ] - Starting openHAB 4.2.1 (Release Build)
...
2024-08-30 22:25:24.103 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:dishwasher:homeconnect:dishwasher1' takes more than 5000ms.
2024-08-30 22:25:29.149 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:api_bridge:homeconnect' takes more than 5000ms.
2024-08-30 22:25:32.449 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.ExceptionInInitializerError: null
	at org.apache.cxf.jaxrs.sse.client.SseEventSourceImpl.connect(SseEventSourceImpl.java:255) ~[?:?]
	at org.apache.cxf.jaxrs.sse.client.SseEventSourceImpl.open(SseEventSourceImpl.java:189) ~[?:?]
	at org.openhab.binding.homeconnect.internal.client.HomeConnectEventSourceClient.registerEventListener(HomeConnectEventSourceClient.java:109) ~[?:?]
	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.lambda$2(AbstractHomeConnectThingHandler.java:312) ~[?:?]
	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.registerEventListener(AbstractHomeConnectThingHandler.java:309) ~[?:?]
	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.lambda$0(AbstractHomeConnectThingHandler.java:148) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException
	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:64) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
	... 13 more
Caused by: java.lang.ClassNotFoundException
	at javax.ws.rs.ext.RuntimeDelegateFinder.newInstance(RuntimeDelegateFinder.java:120) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegateFinder.find(RuntimeDelegateFinder.java:96) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:58) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
	... 13 more
Caused by: java.lang.InstantiationException
	at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
	at java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128) ~[?:?]
	at jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:347) ~[?:?]
	at java.lang.Class.newInstance(Class.java:645) ~[?:?]
	at javax.ws.rs.ext.RuntimeDelegateFinder.newInstance(RuntimeDelegateFinder.java:116) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegateFinder.find(RuntimeDelegateFinder.java:96) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:58) ~[bundleFile:?]
	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
	... 13 more
...
22:26:09.799 [INFO ] [basic.internal.servlet.WebAppServlet] - Stopped Basic UI
2024-08-30 22:26:04.658 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:dishwasher:homeconnect:dishwasher1' takes more than 50
```

I also noticed that, since 4.2, OH seems to be doing parts of the bootup processes twice (e.g., my Rules are initialized twice). Interestingly, there is even a message that "Stopped Basic UI" during the bootup process (no this isn't from a prevous instance). Maybe this is related.

My setup is docker based and fully configured through files.

If there is anything I can do to assist in analyzing the issue, let me know.

**Comment by lsiepel** [comment]
> I'm also seeing this issue since upgrading to 4.2.1 (from 4.1). The warning is printed roughly every 4 minutes.
> 
> What I also noticed are the following message **during bootup** of OH:
> 
> ```
> 2024-08-30 22:23:31.054 [INFO ] [org.openhab.core.Activator          ] - Starting openHAB 4.2.1 (Release Build)
> ...
> 2024-08-30 22:25:24.103 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:dishwasher:homeconnect:dishwasher1' takes more than 5000ms.
> 2024-08-30 22:25:29.149 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:api_bridge:homeconnect' takes more than 5000ms.
> 2024-08-30 22:25:32.449 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
> java.lang.ExceptionInInitializerError: null
> 	at org.apache.cxf.jaxrs.sse.client.SseEventSourceImpl.connect(SseEventSourceImpl.java:255) ~[?:?]
> 	at org.apache.cxf.jaxrs.sse.client.SseEventSourceImpl.open(SseEventSourceImpl.java:189) ~[?:?]
> 	at org.openhab.binding.homeconnect.internal.client.HomeConnectEventSourceClient.registerEventListener(HomeConnectEventSourceClient.java:109) ~[?:?]
> 	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.lambda$2(AbstractHomeConnectThingHandler.java:312) ~[?:?]
> 	at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
> 	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.registerEventListener(AbstractHomeConnectThingHandler.java:309) ~[?:?]
> 	at org.openhab.binding.homeconnect.internal.handler.AbstractHomeConnectThingHandler.lambda$0(AbstractHomeConnectThingHandler.java:148) ~[?:?]
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
> 	at java.lang.Thread.run(Thread.java:840) [?:?]
> Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException
> 	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:64) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
> 	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
> 	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
> 	... 13 more
> Caused by: java.lang.ClassNotFoundException
> 	at javax.ws.rs.ext.RuntimeDelegateFinder.newInstance(RuntimeDelegateFinder.java:120) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegateFinder.find(RuntimeDelegateFinder.java:96) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:58) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
> 	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
> 	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
> 	... 13 more
> Caused by: java.lang.InstantiationException
> 	at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
> 	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
> 	at java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128) ~[?:?]
> 	at jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:347) ~[?:?]
> 	at java.lang.Class.newInstance(Class.java:645) ~[?:?]
> 	at javax.ws.rs.ext.RuntimeDelegateFinder.newInstance(RuntimeDelegateFinder.java:116) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegateFinder.find(RuntimeDelegateFinder.java:96) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:58) ~[bundleFile:?]
> 	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:48) ~[bundleFile:?]
> 	at javax.ws.rs.core.MediaType.valueOf(MediaType.java:101) ~[?:?]
> 	at org.apache.cxf.jaxrs.sse.client.InboundSseEventProcessor.<clinit>(InboundSseEventProcessor.java:47) ~[?:?]
> 	... 13 more
> ...
> 22:26:09.799 [INFO ] [basic.internal.servlet.WebAppServlet] - Stopped Basic UI
> 2024-08-30 22:26:04.658 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'homeconnect:dishwasher:homeconnect:dishwasher1' takes more than 50
> ```
> 
> I also noticed that, since 4.2, OH seems to be doing parts of the bootup processes twice (e.g., my Rules are initialized twice). Interestingly, there is even a message that "Stopped Basic UI" during the bootup process (no this isn't from a prevous instance). Maybe this is related.
> 
> My setup is docker based and fully configured through files.
> 
> If there is anything I can do to assist in analyzing the issue, let me know.

Seems your setup has more issues. It should certainly not have mixed/double start/stop logs.

Could you temporary disable bindings to find out the cause? If all works as expected again, the last step would be to look at this homeconnect binding and log again.

**Comment by Blizzard26** [comment]
I tried disabling a few (most) bindings, but still had the double start/stop in the logs. I then cleared the OH configuration cache at `userdata/config/org/openhab/` and restarted OH. After that I had no more double start/stop in the logs. Also the exception about homeconnect is gone and I haven't seen the "Disposing handler for thing..." warning for the last few hours.

Seems something in my cache was broken....
